### PR TITLE
Check structure.sql and schema.sql

### PR DESCRIPTION
**Pull request by Quiwin**
Add method to check structure.sql/schema.rb and migrations:
- Verify that structure.sql or schema.rb is updated when migrations are added
- Verify structure.sql contains timestamps of new migrations
- Verify schema.rb contains timestamp of last migration

### CODE REVIEW

File: danger-klaxit/lib/danger_plugin.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@ def common
     warn_for_public_methods_without_specs
     warn_for_bad_order_in_config
     warn_rubocop
+    warn_for_not_updated_structure_sql
```
Comment (BuonOmo) [inline danger-klaxit/lib/danger_plugin.rb:22]: this looks quite important to me. What do you think?
```suggestion
    fail_for_not_updated_structure_sql
```

File: danger-klaxit/lib/danger_plugin.rb [hunk 2]
```diff
@@ -119,6 +120,29 @@ def warn_for_bad_order_in_config(config_file = "config/config.yml")
     MARKDOWN
   end
 
+  def warn_for_not_updated_structure_sql
+    migration_files = (git.added_files - %w(Dangerfile)).grep(%r(db/migrate))
```
Comment (BuonOmo) [inline danger-klaxit/lib/danger_plugin.rb:124]: Don't see the point of removing Dangerfile here. bad Copy/pasta 🍝  I guess?
```suggestion
    migration_files = git.added_files.grep(%r(db/migrate))
```

File: danger-klaxit/lib/danger_plugin.rb [hunk 3]
```diff
@@ -119,6 +120,29 @@ def warn_for_bad_order_in_config(config_file = "config/config.yml")
...
+    return nil if migration_files.empty?
+
+    if git.modified_files.grep(%r(db/structure.sql)).empty?
```
Comment (BuonOmo) [inline danger-klaxit/lib/danger_plugin.rb:127]: What about `schema.rb`?
```suggestion
    if git.modified_files.grep(%r(db/(?:structure\.sql|schema\.rb))).empty?
```

Idem for the message!

File: danger-klaxit/lib/danger_plugin.rb [hunk 4]
```diff
@@ -119,6 +120,29 @@ def warn_for_bad_order_in_config(config_file = "config/config.yml")
...
+      warn("You should commit your databases changes via `structure.sql` when you do a migration.")
+      return nil
+    end
+
+    added_migrations_timestamps = migration_files.map do |elt|
```
Comment (BuonOmo) [inline danger-klaxit/lib/danger_plugin.rb:132]: what is elt?

File: danger-klaxit/lib/danger_plugin.rb [hunk 5]
```diff
@@ -119,6 +120,29 @@ def warn_for_bad_order_in_config(config_file = "config/config.yml")
...
+      File.basename(elt).partition("_").first
+    end
+
+    structure_diff = git.diff_for_file("db/structure.sql").patch
+    missing_timestamps = added_migrations_timestamps.reject do |ts|
+      structure_diff.include?(ts)
+    end
+
+    unless missing_timestamps.empty?
+      warn("Some migrations timestamps are missing: #{missing_timestamps.join(", ")}")
```
Comment (BuonOmo) [inline danger-klaxit/lib/danger_plugin.rb:142]: I think this should also be a `fail`!

**Review by BuonOmo — CHANGES_REQUESTED** [review]
Quite nice, still a tiny bit of work imho. Tell me if something's unclear!

**Review by BuonOmo — APPROVED** [review]
LGTM!

**Comment by Quiwin** [comment]
[(4.8) [DANGER] Faire marcher l'alerte de non-modification du structure.sql](https://trello.com/c/KVepTUD1/1127-48-danger-faire-marcher-lalerte-de-non-modification-du-structuresql)

**Pull request merged**